### PR TITLE
Refactor local development environment

### PR DESCRIPTION
**Pull request by esuomi**
These changes are motivated by our need to enable running of tiamat in AWS for [Fintraffic](https://www.fintraffic.fi/en). There's a lot more to come, and these changes work simply as a base for all the upcoming work.

The included changes are:

 - Introduction of `local*` profiles for running the service locally (duh 😄). The core idea is that when developing the service, there should be one lowest friction path to do further feature development on. This is very similar to what's also in [uttu](https://github.com/entur/uttu) which we have worked on previously.
 - `docker-compose.yml` for running supporting services, namely database and also in near future [LocalStack](https://www.localstack.cloud/) to support the development. Here's one obvious confusion I'd like to discuss; there's now `docker-compose.yml` in the root of project, and also in `docker-compose/compose.yml`. This latter in fact confused me at first as I thought it was meant for development, while it really isn't - it's just for running the service locally when tiamat is needed as a dependency for something else or all you want to do is to utilize tiamat's API functionality. Anyway, now that there's multiple files, this opens an avenue for further environment refactoring through [Docker Compose profiles](https://docs.docker.com/compose/profiles/), which we will be using for AWS specific components once we get there development wise.
 - I have no idea how Entur's deployment systems work, so it's probably worth checking that the `application-local.properties` file doesn't conflict with anything on that side.
 - I also took the initiative to fix README.md's general styling a bit, mainly because accurate, type-hinted markdown renders better in GitHub as well.
 - Finally, the local dev ports are all `37xxx`. I simply picked a random number, I use this scheme elsewhere as well where single service and all its dependents gets a port "namespace" of sorts to avoid collisions, as e.g. PostgreSQL in `5432` is _very_ common, and most Java apps tend to use `8080` or `8888`. There's also a wide variety ports already being used in various parts of the documentation and existing configurations, so I wanted to keep this new compose based local environment clearly separated from those.

### CODE REVIEW

File: src/main/resources/application-local.properties [hunk 1]
```diff
@@ -0,0 +1,98 @@
+spring.jpa.database=POSTGRESQL
+spring.datasource.platform=postgres
```
Comment (assadriaz) [inline src/main/resources/application-local.properties:2]: This property is deprecated use spring.sql.init.platform=postgres

File: src/main/resources/application-local.properties [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+spring.jpa.properties.hibernate.order_updates=true
+spring.jpa.properties.hibernate.batch_versioned_data=true
+
+spring.flyway.enabled=true
+spring.flyway.table=schema_version
+
+server.compression.mime-types=application/json,application/xml,text/html,text/xml,text/plain
+
+spring.jpa.hibernate.id.new_generator_mappings=true
+spring.jpa.hibernate.use-new-id-generator-mappings=true
```
Comment (assadriaz) [inline src/main/resources/application-local.properties:20]: Deprecated Hibernate no longer supports disabling the use of new ID generator mappings.

Whether to use Hibernate's newer IdentifierGenerator for AUTO, TABLE and SEQUENCE. This is actually a shortcut for the "hibernate. id. new_generator_mappings" property. When not specified will default to "true".

**Review by assadriaz — COMMENTED** [review]
commented couple of minor changes in application-location.properties, otherwise looks good, tested locally, works out the box.

**Comment by esuomi** [comment]
@assadriaz Thank you for the review, updated accordingly 👍

**Pull request merged**